11-Backbonejs/02-Backbone-Views.md

```diff
@@ -2,145 +2,378 @@
+  },
+  // ...
+];
+```
 
-  // The initialize function is called when you make a new Backbone View.
-var myTodoView = new TodoView();
+Just like before we knew about Backbone, in order to display our data we'll need somewhere in our page to put it. In `build/index.html`, replace the `<div>` with the following HTML:
+
+```html
+<main class="row" id="application">
+  <section class="small-12 columns">
+    <h1>Task List</h1>
+  </section>
+
+  <ul class="task-list small-12 large-6 columns end">
+  </ul>
+</main>
 ```
 
-Inside the curly braces `{}` you place properties for the View object.  One of the most important properties is the `el` property.  The `el` property refers to a DOM object created by the browser.  If you do not define an `el` property Backbone creates one for you which is an empty `div` element.  
+We'll render our tasks as `<li>`s in the `<ul>`.
 
-Below we set the the `el` property to a specific div in the DOM.
+Finally, we need to include Backbone in our project. To do so, add the following line at the top of `app.js`:
 
-```html
-<div id="todo">
-</div>
+```javascript
+import Backbone from 'backbone';
 ```
+
+For the moment, all our JavaScript will go in `src/app.js`. Later on we'll do some organizing, but not yet.
+
+### Adding a View
+Views are created calling `Backbone.View.extend()`. `extend()` is a Backbone thing, and it returns a constructor function we'll use to instantiate our views. If we were in Ruby we'd say we're making a class that inherits from `Backbone.View`. We'll see `extend()` many times in the coming weeks. The only argument to `extend()` is a JavaScript object containing all the extra things we want our view to have.
+
+Backbone can recognize many things in the argument to `extend`, but the two most important for a view are two functions: `initialize()` and `render()`. Let's see what that looks like.
+
 ```javascript
-var TodoView = Backbone.View.extend( {
-  initialize: function() {
-    this.render();
+var TaskView = Backbone.View.extend({
+  initialize: function(options) {
+  },
+
+  render: function() {
+    // Enable chained calls
+    // This is important enough that we'll leave it in, but
+    // we wont talk about it until later.
+    return this;
   }
 });
-
-var myTodoView = new TodoView({ el: $("#todo") });
 ```
 
-### Your First Full View!
+#### Initialize
+Let's start with `initialize()`. This function will be run once when the view is first created. Its job is to get everything ready to go. It takes one argument, `options`, which contains all the stuff the view was created with.
 
-Drawing the view is done in the `render` function which determines how the view displays in the DOM.  Below the
+Views are created with the familiar `new` keyword. It will look something like this:
 
 ```javascript
-var TodoView = Backbone.View.extend({
-  initialize: function(){
-    this.title = "Study Ada Lovelace";
-    this.render();
+var taskView = new TaskView();
```

Comment:
Is this the naming convention we have followed? I'm not sure about using the upper and lowercase variables that are the same, but if this is what we're showing in the examples then this is good.